### ISSUE TITLE
createCountryCollection function throws Match error w/ custom schema

### ISSUE THREAD

**Issue by spencern** [reactioncommerce/reaction#793]
We are extending the ReactionCore.Schema.Shop schema to add a few fields necessary for rental shops. We do this in the following way in one of our packages:

```
ReactionCore.Schemas.RentalShop = new SimpleSchema([
  ReactionCore.Schemas.Shop, {
    rentalShippingBuffer: {
      type: Number,
      optional: true,
      defaultValue: 0
    },
...
  }
]);
```
and then attaching the schema to the Shops collection like so
```
ReactionCore.Collections.Shops.attachSchema(ReactionCore.Schemas.RentalShop);
```

In version v0.12 the `createCountryCollection()` function in core throws a client side Match error 
Uncaught Error: Match error: Match error: rentalShippingBuffer is not allowed by the schema

The error is originating from this line and the error goes away if I change the check to 'Match.any'
https://github.com/reactioncommerce/reaction/blob/development/packages/reaction-core/client/main.js#L348

**Comment by spencern** [comment]
I can fix this issue by adding a line to my package such as:
```
ReactionCore.Schemas.Shop = ReactionCore.Schemas.RentalShop;
```

But I'm unsure if that's a bad practice or will have further repercussions.

**Comment by aaronjudd** [comment]
interesting, yes I guess that would work. but @zenweasel I don't really think we need that check there do we?

**Comment by zenweasel** [comment]
"Need" is a strong word but we should do some type of check. I don't know what "The schema will do it's own check if you attach to the collection...instead of the check" means. 

It seems like some sort of:
```(check, isOfType(shop)) ```

however you might do that in JS

Might be in order.

**Comment by aaronjudd** [comment]
ReactionCore.Collections. Countries.attachSchema(ReactionCore.Schemas.Shops);

**Comment by zenweasel** [comment]
It's my fault, but that still makes no sense to me.

**Comment by zenweasel** [comment]
Closed via PR #794
